src/amp-story-player.js

```diff
@@ -248,15 +377,112 @@ export class AmpStoryPlayer {
    * @param {number} iframeIdx
    */
   displayStory_(iframeIdx) {
-    this.messagingFor_[iframeIdx].sendRequest(
-      'visibilitychange',
-      {state: 'visible'},
-      true
-    );
+    this.handshakePromises_[iframeIdx].then(() => {
+      this.messagingFor_[iframeIdx].sendRequest(
+        'visibilitychange',
+        {state: 'visible'},
+        true
+      );
+    });
+  }
+
+  /**
+   * Sends a message to the story document to pre-render it.
+   * @private
+   * @param {number} iframeIdx
+   */
+  preRenderStory_(iframeIdx) {
+    this.handshakePromises_[iframeIdx].then(() => {
+      this.messagingFor_[iframeIdx].sendRequest(
+        'visibilitychange',
+        {state: 'prerender'},
+        true
+      );
+    });
+  }
+
+  /**
+   * Sends a message to the story document to pause it.
+   * @private
+   * @param {number} iframeIdx
+   */
+  pauseStory_(iframeIdx) {
+    this.handshakePromises_[iframeIdx].then(() => {
+      this.messagingFor_[iframeIdx].sendRequest(
+        'visibilitychange',
+        {state: 'paused'},
```

Comment:
To match the other viewers implementations, we should set the document to `inactive`, and keep `paused` for when the swiping gesture is ongoing.